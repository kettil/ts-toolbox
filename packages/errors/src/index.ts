export { customAggregateError } from './lib/customAggregateError';
export { customError } from './lib/customError';
export { defaultErrors, defaultAggregateErrors } from './lib/defaultErrors';
export { isCustomAggregateError } from './lib/isCustomAggregateError';
export { isCustomError } from './lib/isCustomError';
export { isError } from './lib/isError';
