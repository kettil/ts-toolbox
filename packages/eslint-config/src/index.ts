import { createEslintConfig } from './configs/eslint';
import { createEslintCommentsConfig } from './configs/eslintComments';
import { createFormatjsConfig } from './configs/formatjs';
import { createImportConfig } from './configs/import';
import { createJestConfig } from './configs/jest';
import { createJsxA11yConfig } from './configs/jsxA11y';
import { createNodeConfig } from './configs/node';
import { createReactConfig } from './configs/react';
import { createReactHooksConfig } from './configs/reactHooks';
import { createRegexpConfig } from './configs/regexp';
import { createSonarjsConfig } from './configs/sonarjs';
import { createTestingLibraryConfig } from './configs/testingLibrary';
import { createTestingLibraryJestDomConfig } from './configs/testingLibraryJestDom';
import { createTypescriptConfig } from './configs/typescript';
import { createUnicornConfig } from './configs/unicorn';
import { deepMerge } from './lib/deepEqual';
import { defaultObject } from './lib/defaultObject';
import type { DefaultObject } from './types/defaultObject';

const createConfig = (): Readonly<DefaultObject> => {
  const plugins: readonly DefaultObject[] = [
    createEslintConfig(),
    createNodeConfig(),
    createTypescriptConfig(),

    // common
    createFormatjsConfig(),
    createImportConfig(),
    createUnicornConfig(),
    createEslintCommentsConfig(),
    createSonarjsConfig(),
    createRegexpConfig(),

    // react
    createReactConfig(),
    createReactHooksConfig(),
    createJsxA11yConfig(),

    // testing
    createJestConfig(),
    createTestingLibraryConfig(),
    createTestingLibraryJestDomConfig(),
  ];

  return plugins.reduce<DefaultObject>(
    (previousConfig, plugin) => ({
      env: deepMerge(previousConfig.env, plugin.env, ['env']),
      rules: { ...previousConfig.rules, ...plugin.rules },
      globals: deepMerge(previousConfig.globals, plugin.globals, ['globals']),
      settings: deepMerge(previousConfig.settings, plugin.settings, ['settings']),
      parserOptions: deepMerge(previousConfig.parserOptions, plugin.parserOptions, ['parserOptions']),
      parser: deepMerge(previousConfig.parser, plugin.parser, ['parser']),
      plugins: deepMerge(previousConfig.plugins, plugin.plugins, ['plugins']),
      overrides: deepMerge(previousConfig.overrides, plugin.overrides, ['overrides']),
    }),
    defaultObject,
  );
};

export { createConfig };
