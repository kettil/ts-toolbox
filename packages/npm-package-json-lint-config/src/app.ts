/* eslint-disable @typescript-eslint/naming-convention -- external definition */
import { config as defaultConfig } from './common';
import type { DefaultObject } from './types';

const config: DefaultObject = {
  ...defaultConfig,
  rules: {
    ...defaultConfig.rules,

    'require-publishConfig': 'off',

    'repository-type': 'off',
    'homepage-type': 'off',
    'keywords-type': 'off',

    'valid-values-private': ['error', [true]],
  },
};

export { config };
