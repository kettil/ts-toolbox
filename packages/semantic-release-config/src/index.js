// @semantic-release/changelog
const changelogFile = 'CHANGELOG.md';
const changelogTitle = '# Changelog';
// @semantic-release/git
const assets = [changelogFile, 'package.json', 'package-lock.json'];

const configuration = {
  branches: [
    '+([0-9])?(.{+([0-9]),x}).x',
    'master',
    'main',
    { name: 'beta', prerelease: true },
    { name: 'alpha', prerelease: true },
  ],

  plugins: [
    '@semantic-release/commit-analyzer',
    '@semantic-release/release-notes-generator',
    ['@semantic-release/changelog', { changelogFile, changelogTitle }],
    '@semantic-release/npm',
    ['@semantic-release/git', { assets }],
    '@semantic-release/github',
  ],
  // eslint-disable-next-line no-template-curly-in-string -- @SR requires the notation
  tagFormat: '${version}',
};

module.exports = configuration;
