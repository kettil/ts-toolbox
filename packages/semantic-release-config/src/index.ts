type CreateConfig = () => {
  repositoryUrl?: string;
  tagFormat: string;
  branches: ReadonlyArray<string | { name: string; prerelease?: boolean }>;
  plugins: ReadonlyArray<readonly [string, Record<string, unknown>] | readonly [string]>;
};

// @semantic-release/changelog
const changelogFile = 'CHANGELOG.md';
const changelogTitle = '# Changelog';
// @semantic-release/git
const assets = [changelogFile, 'package.json', 'package-lock.json'];

const createConfig: CreateConfig = () => ({
  branches: [
    '+([0-9])?(.{+([0-9]),x}).x',
    'master',
    'main',
    { name: 'beta', prerelease: true },
    { name: 'alpha', prerelease: true },
  ],

  plugins: [
    ['@semantic-release/commit-analyzer'],
    ['@semantic-release/release-notes-generator'],
    ['@semantic-release/changelog', { changelogFile, changelogTitle }],
    ['@semantic-release/npm'],
    ['@semantic-release/git', { assets }],
    ['@semantic-release/github'],
  ],
  // eslint-disable-next-line no-template-curly-in-string -- @SR requires the notation
  tagFormat: '${version}',
});

export { createConfig };
