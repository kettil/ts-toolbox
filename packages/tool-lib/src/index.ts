export * from './array';
export * from './funcs';
export * from './is';
export * from './node';
export * from './object';
export * from './promise';
export * from './sort';
export * from './string';

export type { Addition } from './types/math/addition';
export type { EqualType } from './types/equalType';
export type { Expect } from './types/expect';
export type { ExtractTypeFromTupleArray } from './types/extractTypeFromTupleArray';
export type { Subtract } from './types/math/subtract';
export type { ObjectClean } from './types/object/objectClean';
export type { ObjectNormalize } from './types/object/objectNormalize';
export type { ObjectType } from './types/object/objectType';
export type { SwaggerExtractProperties } from './types/swagger/swaggerExtractProperties';
export type { SwaggerProperty } from './types/swagger/swaggerProperty';
export type { SwaggerPropertyArray } from './types/swagger/swaggerPropertyArray';
export type { SwaggerPropertyBoolean } from './types/swagger/swaggerPropertyBoolean';
export type { SwaggerPropertyNull } from './types/swagger/swaggerPropertyNull';
export type { SwaggerPropertyNumber } from './types/swagger/swaggerPropertyNumber';
export type { SwaggerPropertyObject } from './types/swagger/swaggerPropertyObject';
export type { SwaggerPropertyString } from './types/swagger/swaggerPropertyString';
export type { Range } from './types/tupel/range';
export type { TupelLength } from './types/tupel/tupelLength';
export type { TupleOf } from './types/tupel/tupleOf';
export type { UnionToTuple } from './types/unionToTuple';
export type { UnionToIntersection } from './types/unionToIntersection';
export type { Unpacked } from './types/unpacked';
