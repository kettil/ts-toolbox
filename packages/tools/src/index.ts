export * from './array';
export * from './date';
export * from './funcs';
export * from './is';
export * from './node';
export * from './object';
export * from './promise';
export * from './sort';
export * from './string';
