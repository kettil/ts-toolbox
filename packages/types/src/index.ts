export * from './assert';
export type { Addition } from './math/addition';
export type { ExtractTypeFromTupleArray } from './common/extractTypeFromTupleArray';
export type { ObjectClean } from './object/objectClean';
export type { ObjectNormalize } from './object/objectNormalize';
export type { ObjectType } from './object/objectType';
export type { Range } from './tupel/range';
export type { Subtract } from './math/subtract';
export type { SwaggerExtractProperties } from './swagger/swaggerExtractProperties';
export type { SwaggerProperty } from './swagger/swaggerProperty';
export type { SwaggerPropertyArray } from './swagger/swaggerPropertyArray';
export type { SwaggerPropertyBoolean } from './swagger/swaggerPropertyBoolean';
export type { SwaggerPropertyNull } from './swagger/swaggerPropertyNull';
export type { SwaggerPropertyNumber } from './swagger/swaggerPropertyNumber';
export type { SwaggerPropertyObject } from './swagger/swaggerPropertyObject';
export type { SwaggerPropertyString } from './swagger/swaggerPropertyString';
export type { TupelLength } from './tupel/tupelLength';
export type { TupleOf } from './tupel/tupleOf';
export type { UnionToIntersection } from './union/unionToIntersection';
export type { UnionToTuple } from './union/unionToTuple';
export type { Unpacked } from './common/unpacked';
