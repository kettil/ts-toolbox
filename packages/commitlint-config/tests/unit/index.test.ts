import { createConfig } from '../../src';

describe('createConfig()', () => {
  test('it should work', () => {
    const config = createConfig();

    expect(config).toMatchSnapshot();
  });
});
