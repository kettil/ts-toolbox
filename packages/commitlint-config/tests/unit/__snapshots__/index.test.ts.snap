// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createConfig() it should work 1`] = `
Object {
  "extends": Array [
    "@commitlint/config-lerna-scopes",
  ],
  "rules": Object {
    "body-leading-blank": Array [
      2,
      "always",
    ],
    "body-max-line-length": Array [
      2,
      "always",
      120,
    ],
    "footer-leading-blank": Array [
      2,
      "always",
    ],
    "footer-max-line-length": Array [
      2,
      "always",
      160,
    ],
    "header-max-length": Array [
      2,
      "always",
      104,
    ],
    "scope-case": Array [
      2,
      "always",
      "lower-case",
    ],
    "subject-case": Array [
      2,
      "never",
      Array [
        "sentence-case",
        "start-case",
        "pascal-case",
        "upper-case",
      ],
    ],
    "subject-empty": Array [
      2,
      "never",
    ],
    "subject-full-stop": Array [
      2,
      "never",
      ".",
    ],
    "type-case": Array [
      2,
      "always",
      "lower-case",
    ],
    "type-empty": Array [
      2,
      "never",
    ],
    "type-enum": Array [
      2,
      "always",
      Array [
        "build",
        "chore",
        "ci",
        "docs",
        "feat",
        "fix",
        "perf",
        "refactor",
        "revert",
        "style",
        "test",
      ],
    ],
  },
}
`;
